external-crates/move/crates/move-vm-runtime/src/tracing2/trace_macros.rs

```diff
@@ -0,0 +1,149 @@
+// Copyright (c) The Move Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+#[macro_export]
+macro_rules! trace_open_main_frame {
+    (
+        $tracer:expr,
+        $args:expr,
+        $ty_args:expr,
+        $function:expr,
+        $loader:expr,
+        $gas_remaining:expr,
+        $link_context:expr
+        $(,)?
+    ) => {
+        // Only include this code when the profiler is enabled
+        #[cfg(feature = "gas-profiler")]
+        $tracer.as_mut().map(|tracer| {
```

Comment:
Instead of CFGing all of these, something you might consider is the following code:

```
#[cfg(feature = "gas-profiler")]
const TRACING_MODE: bool = true;

#[cfg(not(feature = "gas-profiler"))]
const TRACING_MODE: bool = false;
```

Then you can eliminate these macro calls and replace them with:

```
if tracing2::TRACING_MODE {
  tracer.as_mut.map(|tracer| ...)
}
```

Rust will eliminate these blocks even when compiled in debug mode.

In addition, you can even pass the tracer as `Option<&VMTracer>` to avoid the `as_mut`/`map` chains in these macros.